projects/step-frontend/src/lib/modules/timeseries/chart/model/ts-chart-settings.ts

```diff
@@ -8,13 +8,14 @@ export interface TSChartSettings {
   series: TSChartSeries[];
   scales?: Scale[];
   autoResize?: boolean; // autoresize on window size change
-  axes?: Axis[];
+  axes: Axis[];
   showLegend?: boolean; // show legend behind the chart
   yScaleUnit?: string; // the unit which will be displayed along with the Y axis.
   zScaleTooltipLabel?: string;
 }
 
 export interface TSChartSeries extends Series {
-  data: (number | null | undefined)[];
   id: string;
+  data: (number | null | undefined)[];
+  legendName?: string; // id is used if not specified.
```

Comment:
Are we using this anywhere without specifying the name? I would prefer to make legendName mandatory and if we have no name somewhere to specify the id as the parameter when instanciating this

Reply:
there may be charts which don't have the legend (for example the ranger), so that's why I put it optionally. I will change it to mandatory